server/src/main/java/org/elasticsearch/common/unit/Processors.java

```diff
@@ -137,7 +137,7 @@ public int roundDown() {
     }
 
     private static boolean validNumberOfProcessors(double processors) {
-        return Double.isFinite(processors) && processors > 0.0;
+        return Double.isFinite(processors) && processors >= 0.0;
```

Comment:
Judging by the rest of this file zero should be allowed.

E.g. 
- the constructor starts with `if (count == 0.0)` (line 42)
- the exception message reads `processors must be a non-negative number` (line 59)

Reply:
This is causing the `/_ml/info` endpoint to fail when no ML nodes are started.

See also (private Slack channel): https://elastic.slack.com/archives/G2FUG2B61/p1701184000628989